mindmeld/app.py

```diff
@@ -161,6 +168,97 @@ def add_dialogue_rule(self, name, handler, **kwargs):
         else:
             self._dialogue_rules.append((name, handler, kwargs))
 
+    def custom_action(self, **kwargs):
+        """Adds a custom action, similar to `add_custom_action` but allows the ordering
+        of nlp entities to be more flexible.
+
+        Examples:
+            app.custom_action(intent='greeting', action='say_greeting')
+            app.custom_action(entity='person', action='greet_person')
+        """
+        action = kwargs.pop("action", None)
+        actions = kwargs.pop("actions", None)
+
+        if not (action or actions):
+            raise CustomActionException(
+                "`action` or `actions` must be present in arguments."
+            )
+
+        if action:
+            self.add_custom_action(
+                action,
+                asynch=kwargs.pop("asynch", False),
+                overwrite=kwargs.pop("overwrite", False),
+                config=kwargs.pop("config", None),
+                **kwargs
+            )
+        else:
+            self.add_custom_action_sequence(
+                actions,
+                asynch=kwargs.pop("asynch", False),
+                overwrite=kwargs.pop("overwrite", False),
+                config=kwargs.pop("config", None),
+                **kwargs
+            )
+
+    def add_custom_action(
+        self, action, asynch=False, overwrite=False, config=None, **kwargs
```

Comment:
asynch -> async

Reply:
`async` is a Python keyword so you can't use it; it can be either `asynch` or `async_mode`. I chose `asynch` because it is shorter.

Reply:
elsewhere in the code we have been using `async_mode`. I think it's better to be consistent with that.

Reply:
Done